### PR TITLE
Fix a bug with samples from only one track being flushed once the last pad is connected in ISOM demuxer

### PR DESCRIPTION
**Pull request by varsill**

### CODE REVIEW

File: lib/membrane_mp4/demuxer/isom.ex [hunk 1]
```diff
@@ -460,15 +460,16 @@ defmodule Membrane.MP4.Demuxer.ISOM do
     Range.size(tracks) == length(pads)
   end
 
-  defp flush_samples(state, track_id) do
-    buffers =
-      Map.get(state.buffered_samples, track_id, [])
-      |> Enum.reverse()
-      |> Enum.map(fn {buffer, ^track_id} -> buffer end)
+  defp flush_samples(state) do
+    Enum.reduce(Map.keys(state.buffered_samples), {[], state}, fn track_id, {actions, state} ->
```
Comment (mat-hek) [inline lib/membrane_mp4/demuxer/isom.ex:464]: use flat_map_reduce

Comment (varsill) [reply lib/membrane_mp4/demuxer/isom.ex:464]: done

File: lib/membrane_mp4/demuxer/isom.ex [hunk 2]
```diff
@@ -460,15 +460,16 @@ defmodule Membrane.MP4.Demuxer.ISOM do
...
+      buffers =
+        Map.get(state.buffered_samples, track_id, [])
```
[the remark this answers is not in the document]
Comment (varsill) [reply lib/membrane_mp4/demuxer/isom.ex:466]: indeed, I have just patched the exisiting `flush_samples` solution and I haven't thought of this :D

[the remark this answers is not in the document]
Comment (varsill) [reply lib/membrane_mp4/demuxer/isom.ex:466]: done

File: lib/membrane_mp4/demuxer/isom.ex [hunk 3]
```diff
@@ -460,15 +460,16 @@ defmodule Membrane.MP4.Demuxer.ISOM do
...
+      buffers =
+        track_samples
+        |> Enum.reverse()
+        |> Enum.map(fn {buffer, ^track_id} -> buffer end)
 
-    actions = [buffer: {Pad.ref(:output, track_id), buffers}]
-
-    {actions, put_in(state, [:buffered_samples, track_id], [])}
+      new_actions = [buffer: {Pad.ref(:output, track_id), buffers}]
+      {new_actions, put_in(state, [:buffered_samples, track_id], [])}
```
Comment (mat-hek) [inline lib/membrane_mp4/demuxer/isom.ex:471]: NIT: I'd reset buffered samples in another run, so this actually could become a flat_map

Comment (mat-hek) [inline lib/membrane_mp4/demuxer/isom.ex:471]:
```suggestion
     actions = [buffer: {Pad.ref(:output, track_id), buffers}]
     {actions, put_in(state, [:buffered_samples, track_id], [])}
```

Comment (varsill) [reply lib/membrane_mp4/demuxer/isom.ex]: done

Comment (varsill) [reply lib/membrane_mp4/demuxer/isom.ex]: I have used `map`

**Review by mat-hek — APPROVED** [review]

**Pull request merged**